src/hyphenClient.ts

```diff
@@ -1,33 +1,34 @@
-import type { EvaluationResponse, HyphenEvaluationContext } from './types';
-import NodeCache from '@cacheable/node-cache';
-import { horizon, cache } from './config';
+import type { EvaluationResponse, HyphenEvaluationContext, HyphenProviderOptions } from './types';
+import { horizon } from './config';
 import type { Logger } from '@openfeature/server-sdk';
+import { CacheClient } from './cacheClient';
 
 export class HyphenClient {
   private readonly publicKey: string;
   private readonly horizonServerUrls: string[];
-  private cache: NodeCache;
+  private cache: CacheClient;
 
-  constructor(publicKey: string, horizonServerUrls: string[] = []) {
+  constructor(publicKey: string, options: HyphenProviderOptions) {
     this.publicKey = publicKey;
+
+    const horizonServerUrls = options.horizonServerUrls || [];
     horizonServerUrls.push(horizon.url);
     this.horizonServerUrls = horizonServerUrls;
-    this.cache = new NodeCache({
-      stdTTL: cache.ttlSeconds,
-      checkperiod: cache.ttlSeconds * 2,
-    });
+
+    this.cache = new CacheClient(options.cache);
   }
 
   async evaluate(context: HyphenEvaluationContext, logger?: Logger): Promise<EvaluationResponse> {
-    const cachedResponse = this.cache.get<EvaluationResponse>(context.targetingKey);
+    const cacheKey = this.cache.generateCacheKey(context);
+    const cachedResponse = this.cache.get<EvaluationResponse>(cacheKey);
```

Comment:
if you are going through the trouble of making specific cache.get and cache.set function beyond the defaults then you should change the signiture to match your needs. 

So the this get fuction should take the context and handle calculating the key and not make the consumer do both.

Reply:
Yep, I was doing this because I was trying to expose set and get functions as is.